pyvirtualcam/native_macos/main.mm

```diff
@@ -0,0 +1,165 @@
+// Rest of files is stolen from:
+// - https://github.com/obsproject/obs-studio/tree/516ed4458dbc74b27b425533789d4e9dd95b923b/plugins/mac-virtualcam
+// - https://github.com/descampsa/yuv2rgb/tree/c51a543cab4322598047bcd86f5d5fd680e9e6f2
+// Consider migration to https://github.com/johnboiles/coremediaio-dal-minimal-example ?
+
+#include <stdexcept>
+#include <pybind11/pybind11.h>
+#include <pybind11/numpy.h>
+#include "OBSDALMachServer.h"
+#include "Defines.h"
+#include <mach/mach_time.h>
+#include <cstdint>
+
+namespace py = pybind11;
+
+#define UNUSED_PARAMETER(arg)
+
+
+static OBSDALMachServer *sMachServer = nil;
+
+static int cam_output_width;
+static int cam_output_height;
+static uint32_t cam_fps_num;
+static uint32_t cam_fps_den;
+
+
+static void YfromRGB(uint8_t* y, uint8_t r, uint8_t g, uint8_t b) {
+    *y = (uint8_t)( 0.257 * r + 0.504 * g + 0.098 * b +  16);
+}
+static void UVfromRGB(uint8_t* u, uint8_t* v, uint8_t r, uint8_t g, uint8_t b) {
+    *u = (uint8_t)(-0.148 * r - 0.291 * g + 0.439 * b + 128);
+    *v = (uint8_t)( 0.439 * r - 0.368 * g - 0.071 * b + 128);
+}
+
+static bool virtual_output_start(int width, int height, double fps, int delay) {
```

Comment:
`delay` argument is being ignored. What's the intention here?

Reply:
It determines the size of the ring buffer for the Windows variant, roughly small delay = small ring buffer = low latency = potentially more instable. Kind of an advanced option. It's fine to leave it unused here.

Reply:
I assumed that was the case.

If you wanted support for this, you could probably play with the presentation-timestamp on macOS to achieve something similar (pre-buffer multiple frames).
I currently use the `mach_absolute_time` function to create a timestamp, but I'm not sure if that has any impact (`mach_absolute_time` is something I saw mentioned in the OBS plugin, although I did not verify it's the correct timestamp source; and then the DAL plugin might ignore these timestamps anyway).